Task-46959 : Analytics issue when uploading and updating file

AzmiTouil opened the pull request:
Prior to this change, the statistic considers the property changes related to the editor as update operation.
To fix this problem, we need to exclude these properties

houssemexo26 committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

GouadriaHanen committed:
TASK-46944 : fixed displaying wrong week number (#163)

in analytics when choosing  'X AXIS' tab in the top menu and check 'Per week' it displays a wrong week number (actual week number +1)
since the DateTimeFormatter.ofPattern(' 'W' w uuuu') returns  week-based date without an offset
fixed by formatting date with 'BASIC_ISO_DATE' that return the right week number without offset

AzmiTouil committed:
Task-46959 : Analytics issue when uploading and updating file
Prior to this change, the statistic considers the property changes related to the editor as update operation.
To fix this problem, we need to exclude these properties

ahamdi commented on analytics-listeners/src/main/java/org/exoplatform/analytics/listener/jcr/JCRNodeListener.java:
```diff
@@ -80,6 +81,16 @@
 
   private static final Set<String>              CURRENTLY_PROCESSING_NODE_PATH_QUEUE = new HashSet<>();
 
+  private static final String                   EXO_USER_PREFERENCES                 = "exo:userPrefferences";
+
+  private static final String                   EXO_EDITORS_RUNTIME_ID               = "exo:editorsId";
+
+  private static final String                   EXO_CURRENT_PROVIDER                 = "exo:currentProvider";
+
+  private static final String                   EXO_PREFERRED_EDITOR                 = "exo:prefferedEditor";
+
+  private static final List<String>             excludedPropertyNames                = List.of(EXO_EDITORS_RUNTIME_ID, EXO_CURRENT_PROVIDER, EXO_PREFERRED_EDITOR);
+
```
We may have other properties to ignore later, let's make this possible by configuration. We should not hardcode them in this Listener

GouadriaHanen committed:
TASK-48842 : fix Wrong Connections count number when user connects many time in the same day (#168)

when a user login each time the login listener indexes a new distinct doc to the ES.
while counting the number of connections it gets all the login docs since are DISTINCT
fixed by avoiding adding a doc if he has already logged in the same day
fixed getting lastConnectionTime table field from the already retrieved identity

AzmiTouil committed:
Task-46959: PR Review

ahamdi commented on analytics-listeners/src/main/java/org/exoplatform/analytics/listener/jcr/JCRNodeListener.java:
```diff
@@ -110,7 +131,9 @@ public boolean execute(Context context) throws Exception { // NOSONAR
         return true;
       }
-
+      if (eventType == Event.NODE_ADDED && node.getPrimaryNodeType().getName().equals(EXO_USER_PREFERENCES)) {
```
Use node.isNodeType()

ahamdi commented on analytics-listeners/src/main/java/org/exoplatform/analytics/listener/jcr/JCRNodeListener.java:
```diff
@@ -100,6 +108,19 @@ public boolean execute(Context context) throws Exception { // NOSONAR
+      String excludedProperty = PropertyManager.getProperty(EXCLUDED_PROPERTY_NAMES);
+      if (StringUtils.isNotBlank(excludedProperty)) {
+        String[] excludedPropertyNames = excludedProperty.split(",");
+        if (eventType == Event.PROPERTY_ADDED || eventType == Event.PROPERTY_CHANGED) {
+          PropertyImpl property = (PropertyImpl) context.get(InvocationContext.CURRENT_ITEM);
```
```suggestion
        if (item instanceof Property && (eventType == Event.PROPERTY_ADDED || eventType == Event.PROPERTY_CHANGED)) {
```

ahamdi reviewed (changes requested):
A note in documentation should be added

AzmiTouil committed:
Task-46959: Update PR

ahamdi committed:
Merge branch 'feature/maintenance' into task-46959

ahamdi commented on analytics-listeners/src/main/resources/conf/portal/configuration.xml:
```diff
@@ -325,6 +325,9 @@
                       <string>addNode,addProperty,changeProperty,removeProperty</string>
                     </field>
+                    <field name="ignoredProperties">
+                      <string>${exo.analytics.ignored-property.name:exo:editorsId,exo:currentProvider,exo:prefferedEditor}</string>
```
```suggestion
                      <string>${exo.analytics.documents.ignored-properties:exo:editorsId,exo:currentProvider,exo:prefferedEditor}</string>
```

AzmiTouil committed:
Task-46959: PR Review
